### DISCUSSION TITLE
What should happen when certain locale does not have data?

### DISCUSSION THREAD

**Discussion by pkuczynski** [faker-js/faker#1198] [Polls]
Umbrella issue: https://github.com/faker-js/faker/issues/893

**Comment by pkuczynski** [comment]
Would someone using Chinese be happy to see English strings for missing data?

**Reply by ST-DDT** [reply]
IMO the answer is it depends.
If the locale data contain it, then it should return it (regardless of whether it is "fallback" or the "main" locale).
If I don't want a fallback to English, then I won't set it as such. I prefer configuration over assumption.
It should be possible to check whether the data exist without throwing an error. e.g. `faker.extensions` return undefined if the entry is missing
It should be possible to throw an error when the requested data are missing without extra checks (e.g. in most of our methods). e.g. `faker.definitions` throw if the entry is missing
